cartridges/int_algolia/cartridge/scripts/services/algoliaServiceHelper.js

```diff
@@ -143,6 +146,70 @@ function callJsonService(title, service, params) {
     return callStatus;
 }
 
+/**
+ * Validates API key permissions by checking ACLs and index access
+ * @param {dw.svc.Service} service - Service instance to call
+ * @param {string} applicationID - Algolia Application ID
+ * @param {string} adminApiKey - Algolia Admin API Key
+ * @returns {Object} Response indicating validation success or failure
+ */
+function validateAPIKey(service, applicationID, adminApiKey) {
+    const keyResponse = service.call({
+        method: 'GET',
+        url: 'https://' + applicationID + '.algolia.net/1/keys/' + adminApiKey
+    });
+
+    if (!keyResponse.ok) {
+        return {
+            error: true,
+            errorMessage: Resource.msg('algolia.error.key.validation', 'algolia', null)
+        };
+    }
+
+    const keyData = keyResponse.object.body;
+    const requiredACLs = ['addObject', 'deleteObject', 'deleteIndex', 'settings'];
+    const missingACLs = requiredACLs.slice();
+    const excessiveACLs = [];
+
+    keyData.acl.forEach(function(acl) {
+        const aclIndex = missingACLs.indexOf(acl);
+        if (aclIndex === -1) {
+            excessiveACLs.push(acl);
+        } else {
+            missingACLs.splice(aclIndex, 1);
+        }
+    });
+
+    if (missingACLs.length > 0) {
+        const errorMessage = Resource.msgf('algolia.error.missing.permissions', 'algolia', null, missingACLs.join(', '));
+        return {
+            error: true,
+            errorMessage: errorMessage,
+            warning: excessiveACLs.length > 0 ? Resource.msgf('algolia.warning.excessive.permissions', 'algolia', null, excessiveACLs.join(', ')) : ''
+        };
+    }
+
+    const testIndex = algoliaData.calculateIndexName('products');
```

Comment:
This leads to having the BM completely stuck and not being able to save settings anymore, with the following scenario:

- Add an index prefix, e.g. "test"
- Save the settings. It works, because `calculateIndexName` methods **uses the preference already saved (no prefix so it fallback to the default one)**. In my case, the `testIndex` is set to `zzgk_008_dx__RefArch__products__en_US`. And it works only **because this index exists in my App**.
- Remove the index prefix
- Save the settings: it now **permanently fails**. Because the `IndexPrefix` preference is now set to `"test"` and I don't have any `test__products__en_US` index in Algolia, the call to Algolia returns 404. I am not able to save settings anymore.